### ISSUE TITLE
SSI database credentials

### ISSUE THREAD

**Issue by emanuele45** [elkarte/Elkarte#3258]
Working on the database for the PSR4 I broken up the classes and reworked most of the code.
I'm now working on the initialization of the connection, and I'm wondering: what the heck is the purpose of the SSI credentials?
I mean: if we use another connection for SSI, this connection should have access to the original database anyway, because we **need** all the content of the members table, permissions, etc.
There is quite a bit of stuff we cannot do without.
So, we'd need to somehow swap database at some points in the SSI-related code in order to keep the queries point to the right databases.

From the use-cases I can think of, this solution honestly is not really something that you see around every day.
The way I'm reworking the code would allow to instantiate multiple connections (give the fact I'm dropping the singleton), so I think that having database return the connection to "our" database and nothing else. Then, if someone needs a connection to a different db, he can instantiate a new connection and have its own "database" function (maybe with a different name) and do it's work.

Bottom line: I propose to remove any reference to SSI-related connection databases and live with just one connection inside our code.

What do you think?

**Comment by Spuds** [comment]
The only thing I can think of would be to allow a less privileged db user access via SSI interface.

**Comment by Vekseid** [comment]
The only real reason to do that is to separate the writer and reader. My most complex SSI code has to bring in the real credentials anyway.
